packages/@guardian/source-foundations/src/utils/jest-matchers/toBeValidCSS.ts

```diff
@@ -0,0 +1,53 @@
+import type { Warning } from 'lightningcss';
+import { transform } from 'lightningcss';
+
+type ToBeValidCSSOptions = {
+	/**
+	 * Set this to true if the CSS is a fragment (e.g. not wrapped in a selector and valid on its own)
+	 */
+	isFragment?: boolean;
+};
+
+declare global {
+	// eslint-disable-next-line @typescript-eslint/no-namespace -- we extend the jest matchers interface like this
+	namespace jest {
+		interface Matchers<R> {
+			toBeValidCSS(options?: ToBeValidCSSOptions): R;
+		}
+	}
+}
+
+expect.extend({
+	/**
+	 * Uses the lightningcss library to validate given CSS
+	 *
+	 * @param received - The CSS to validate
+	 * @param options - Specify whether the CSS provided is a fragment (not wrapped in a selector)
+	 */
+	toBeValidCSS(
+		received: string,
+		options: ToBeValidCSSOptions = {},
+	): jest.CustomMatcherResult {
+		const { isFragment = false } = options;
+
+		// We wrap the CSS in a selector if it is a fragment to ensure it is valid.
+		const finalCSS = isFragment ? `* { ${received} }` : received;
+
+		try {
+			transform({
+				code: Buffer.from(finalCSS, 'utf8'),
+				filename: '',
+			});
+
+			return {
+				pass: true,
+				message: () => '',
+			};
+		} catch (e) {
+			return {
+				pass: false,
+				message: () => (e as Warning).message,
+			};
+		}
```

Comment:
You can probably check that `e instanceof Warning` and otherwise throw that again.

```ts
		catch (e: unknown) {
			if(!(e instanceof Warning) throw e;
			return {
				pass: false,
				message: () => (e as Warning).message,
			};
		}
```

Reply:
couldn't check to see if it was a `Warning` because it's only supplied as an interface which we can't do an `instanceof` check against.

Instead I've put a check in for the `message` property and throw if that is not present